doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+| binaryblob        | binary (0x05)       | length                                             |
+| datetime          | UTC datetime (0x09) | 8                                                  |
+| doublearray       | Array (0x04)        | (8 + keysize) * count                              |
+| integerarray      | Array (0x04)        | (4 + keysize) * count                              |
+| booleanarray      | Array (0x04)        | (1 + keysize) * count                              |
+| longintegerarray  | Array (0x04)        | (1 + keysize) * count                              |
+| stringarray       | Array (0x04)        | depends on count, length, keys length and encoding |
+| binaryblobarray   | Array (0x4)         | depends on count, keys length and length           |
+
+Object aggregations are encoded as embedded documents.
+
+## Connection and Disconnection
+
+A device is not required to publish any additional connection or disconnection message, the MQTT broker automatically keeps track of those events.
+When connecting, before publishing any data message, a device should check MQTT _session present_ flag. When _session present_ is _true_ no further actions are required, when _false_ the device should take following actions:
+
+* Publish introspection
+* Publish empty cache
+* Publish all set properties on all property interfaces
+
+If a device is unable to inspect _session present_ all previous actions must be taken at every reconnection.
+
+## Introspection
+
+Each device must declare the set of supported interfaces and their version. Astarte needs to know the interfaces before processing any further data publish.
+This message in Astarte jargon is called _introspection_ and it's performed by publishing on the device root topic the list of supported interfaces.
+
+Introspection payload is a simple plain text string, and it has the following format (in BNF like syntax):
+
+```
+introspection ::= introspection_list
+introspection_list ::= introspection_entry ";" introspection_list | introspection_entry
+introspection_entry ::= interface_name ":" interface_major_version ":" interface_minor_version
+```
+
+The following example is a valid introspection payload:
+
+```
+com.example.MyInterface:1:0;org.example.DraftInterface:0:3
+```
+
+## Empty Cache
+
+A device may declare to have no knowledge about remote set properties. This usually happens on a new device, after a factory reset or when cache is lost.
+In Astarte jargon this message is called _empty cache_ and it is performed by publising "1" on the device `/control/emptyCache` topic.
+
+After an empty cache message properties might be purged and Astarte might publish all the server owned properties again.
+
+## Session Present
+
+Astarte may declare to have no (or inconsistent) knowledge about device owned properties, this might happen after an error (Astarte should keep knowledge about a remote device status).
```

Comment:
Rephrase:

In the very same fashion as the device, Astarte (or the broker) might be inconsistent with a Device's known status and its known properties. Although unlikely, as Astarte should always keep knowledge about remote device status, this might happen, for example, after an internal error.